multimodal-search-pdf/pods/segmenters.py

```diff
@@ -0,0 +1,285 @@
+import io
+from typing import Dict, List
+
+import numpy as np
+from jina.executors.segmenters import BaseSegmenter
+from jina.executors.decorators import single, batching
+
+
+class PDFSegmenter(BaseSegmenter):
+    """
+    :class:`PDFExtractorSegmenter` Extracts data (text and images) from PDF files.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    @single(slice_nargs=3)
+    def segment(self, uri: str, buffer: bytes, mime_type: str, *args, **kwargs) -> List[Dict]:
+        """
+        Segements PDF files. Extracts data from them.
+
+        Checks if the input is a string of the filename,
+        or if it's the file in bytes.
+        It will then extract the data from the file, creating a list for images,
+        and text.
+
+        :param uri: File name of PDF
+        :type uri: str
+        :param buffer: PDF file in bytes
+        :type buffer: bytes
+        :param mime_type: the type of data
+        :returns: A list of documents with the extracted data
+        :rtype: List[Dict]
+        """
+        chunks=[]
+        if mime_type == 'application/pdf':
+            import fitz  # fitz is a library used in `PyMuPDF` to read pdf and images
```

Comment:
A long code block between the `if` clause and the `return` makes the codes difficult to read
```suggestion
        if mime_type != 'application/pdf':
            return chunks
        import fitz  # fitz is a library used in `PyMuPDF` to read pdf and images
```